File: tests/grader/fixtures/problems_for_verification_tests/hworld/solutions/chris-csharp-t/Hworld.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading;

namespace Hworld
{
    internal class InputData
    {
        public int Count { get; private set; }

        public InputData(int count)
        {
            Count = count;
        }
    }

    internal class OutputData
    {
        public List<String> Lines { get; private set; }
        
        public OutputData(List<String> lines)
        {
            Lines = lines;
        }
    }
    
    internal class Hworld
    {
        public static void Main(string[] args)
        {
            string inputFileName = args.Length == 0 ? "hworld.in" : args[0];
            string outputFileName = args.Length == 0 ? "hworld.out" : args[1];
            
            InputData inputData = ReadInput(inputFileName);
            OutputData outputData = Solve(inputData);
            WriteOutput(outputFileName, outputData); 
        }
        
        private static InputData ReadInput(string fileName)
        {
            int count = int.Parse(File.ReadAllText(fileName));
            return new InputData(count);
        }
        
        private static OutputData Solve(InputData inputData)
        {
            Thread.Sleep(TimeSpan.FromSeconds(5));
            var lines = Enumerable.Range(0, inputData.Count).Select(i => "Hello world!").ToList(); 
            return new OutputData(lines);
        }
        
        private static void WriteOutput(string fileName, OutputData outputData)
        {
            File.WriteAllLines(fileName, outputData.Lines);
        }
    }
}

File: tests/grader/fixtures/problems_for_verification_tests/hworld/solutions/jane-java-e/Hworld.java
import java.io.*;
import java.util.*;

public class Hworld {

    public static void main(String[] args) throws IOException {
        String inputFileName = args.length >= 2 ? args[0] : "hworld.in";
        String outputFileName = args.length >= 2 ? args[1] : "hworld.out";

        HworldInputData inputData = readInput(inputFileName);
        HworldOutputData outputData = solve(inputData);
        writeOutput(outputFileName, outputData);
    }

    private static HworldInputData readInput(String inputFileName) throws IOException {
        File inputFile = new File(inputFileName);
        try (FileReader inputFileReader = new FileReader(inputFile)) {
            try (BufferedReader bufferedReader = new BufferedReader(inputFileReader)) {
                int count = Integer.valueOf(bufferedReader.readLine());
                return new HworldInputData(count);
            }
        }
    }

    private static HworldOutputData solve(HworldInputData inputData) {
        String lines = Collections.nCopies(inputData.getCount(), "Hello world!");
        return new HworldOutputData(lines);
    }

    private static void writeOutput(String outputFileName, HworldOutputData outputData) throws IOException {
        try (Writer outputFileWriter = new FileWriter(outputFileName)) {
            for (String line: outputData.getLines()) {
                outputFileWriter.write(line);
                outputFileWriter.write("\n");
            }
        }
    }
}
